### PR TITLE
API-26949: DQ Update Medication Reference

### PR DESCRIPTION
**Pull request by ssimonian-va**
Update MedicationReference field in MedicationRequest/Order/Statement

MedicationReference includes the following populated fields for fingerprint use only:
- reference url
- display
- identifier.system
- identifier.value

Layout:
```
"medicationReference": {
    "reference": "https://sandbox-api.va.gov/services/fhir/v0/r4/Medication/I2-OYQT7HD7CEATIUNPD4M6D27LXM000000",
    "display" : "Loratadine 5 MG Chewable Tablet"
    "identifier" : [{
        "system": "http://www.nlm.nih.gov/research/umls/rxnorm",  
        "value": 665078
        }]
}
```

### CODE REVIEW

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationOrderTransformerTest.java [hunk 1]
```diff
@@ -159,12 +161,20 @@ void statusTests() {
 
   @Test
   void toFhir() {
-    DatamartMedicationOrder dmo = MedicationOrderSamples.Datamart.create().medicationOrder();
+    DatamartMedicationOrder dmMedOrder = MedicationOrderSamples.Datamart.create().medicationOrder().rxNorm("123456");
+    var sampleMedRequest = MedicationRequestSamples.R4.create().medicationRequestFromMedicationOrder();
```
Comment (ilaflamme-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationOrderTransformerTest.java:165]: Same comment as above.

**Comment by ilaflamme-va** [comment]
I've tested this locally. The new structure is looking good to me. I see RXNORM's populating for our mock patients.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationorder/Dstu2MedicationOrderTransformer.java [hunk 2]
```diff
@@ -134,6 +138,20 @@ private Duration duration(Optional<Integer> maybeValue) {
     return null;
   }
 
+  Reference medReference(DatamartReference medication) {
+    var reference = Dstu2Transformers.asReference(medication);
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationorder/Dstu2MedicationOrderTransformer.java:142]: keep the static import

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationorder/Dstu2MedicationOrderTransformer.java [hunk 3]
```diff
@@ -134,6 +138,20 @@ private Duration duration(Optional<Integer> maybeValue) {
...
+    if (reference != null && datamart.rxNorm() != null) {
+      reference.extension(
+          Extension.builder()
+              .valueIdentifier(
+                  Identifier.builder().system(FhirSystems.RXNORM).value(datamart.rxNorm()).build())
+              .url("rxnorm")
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationorder/Dstu2MedicationOrderTransformer.java:148]: populate `url`, then `value` to match the order of fields in the spec

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformer.java [hunk 4]
```diff
@@ -5,24 +5,29 @@
...
 import java.time.Instant;
 import java.util.List;
 import java.util.Optional;
 import lombok.Builder;
 import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
 
 /** Transform a datamart medication statement to an r4 fhir medication request. */
 @Builder
+@Slf4j
```
[the remark this answers is not in the document]
Comment (ssimonian-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformer.java:30]: I believe so. @ilaflamme-va can you confirm?

[the remark this answers is not in the document]
Comment (jblefkove-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medicationrequest/R4MedicationRequestFromMedicationStatementTransformer.java:30]: I don't see any logging in this file... I'll go ahead and remove this

**Pull request merged**